presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -408,6 +423,10 @@ protected Boolean visitDereferenceExpression(DereferenceExpression node, Void co
 
         private Boolean isField(QualifiedName qualifiedName)
         {
+            if (lambdaVariableNames.contains(qualifiedName.toString())) {
```

Comment:
Thank you @findepi and @sopel39 for your comments.

`columnReferences` doesn't work because it's not an identity set. I went with second suggestion from @sopel39